server/common/src/main/java/org/eclipse/lsp/cobol/common/copybook/CopybookService.java

```diff
@@ -73,4 +70,11 @@ ResultWithErrors<CopybookModel> resolve(
    * @param processingMode copybook processing mode.
    */
   void sendCopybookDownloadRequest(String documentUri, Collection<String> copybookUris, CopybookProcessingMode processingMode);
+
+  /**
+   *  Pre-processed a copybookModel and returns a cleaned copybook with and associated errors
+   * @param dirtyCopybook copybook Model to be pre-processed
+   * @return A pre-processed copybook with and associated errors
+   */
+  ResultWithErrors<CopybookModel> cleanupCopybook(CopybookModel dirtyCopybook);
```

Comment:
Reason - In case a dialect has its own way of getting a copybook content. We should give a mechanism to pre-process it.
For Db2Dialect, it has some pre-defined copybook and dialect would like to clean it up before extending it.